api/v1beta1/disruption_types.go

```diff
@@ -531,3 +530,44 @@ func ShouldSkipNodeFailureInjection(disKind chaostypes.DisruptionKindName, insta
 	// we should never re-inject a static node failure, as it may be targeting the same pod on a new node
 	return disKind == chaostypes.DisruptionKindNodeFailure && instance.Spec.StaticTargeting && injection.InjectionStatus != chaostypes.DisruptionInjectionStatusNotInjected
 }
+
+// TargetedContainers returns a map of containers with containerName as a key and containerID in the format '<type>://<container_id>' as a value
+func TargetedContainers(pod *corev1.Pod, containerNames []string) (map[string]string, error) {
```

Comment:
Why keep this function in this file?

Reply:
Utils lack intent and expresiveness IMO, I'm planning to remove it soon 😆 

This function should definitely be part of our API specs, how to retrieve targeted containers for a given disruption, it's a business definition, hence should live here.

Reply:
> Utils lack intent and expresiveness IMO, I'm planning to remove it soon 😆

Sure, I agree with that

> This function should definitely be part of our API specs, how to retrieve targeted containers for a given disruption, it's a business definition, hence should live here.

At least for now it isn't disruption specific, it's just looking for names within pods, but there isn't a misc k8s utils file, so this works. 👍

Reply:
Because I wrongly removed it from DisruptionSpec.

In fact it's related to a disruption spec:
* when no container available, retrieve all
* when container already targeted, return the ones targeted

I faced some issues with tests and decided to change the signature, I will get back to the "good" one, now that I believe I finally found the root cause of all my tests issues 😬 PEBCAK